src/Symfony/Component/Security/Core/Tests/Authentication/Token/AbstractTokenTest.php

```diff
@@ -96,6 +97,19 @@ public function testSerialize()
         $this->assertEquals($token->getAttributes(), $uToken->getAttributes());
     }
 
+    public function testSerializeWithRoleObjects()
+    {
+        $user = new User('name', 'password', array(new Role('ROLE_FOO'), new Role('ROLE_BAR')));
+        $token = new ConcreteToken($user, $user->getRoles());
```

Comment:
Point is; passing roles here is effectively the same as passing in roles via `$this->getToken($roles)`.. like above.